rust/vectordb/src/query.rs

```diff
@@ -64,17 +76,24 @@ impl Query {
         }
     }
 
-    /// Execute the queries and return its results.
+    /// Convert the query plan to a [`DatasetRecordBatchStream`]
     ///
     /// # Returns
     ///
     /// * A [DatasetRecordBatchStream] with the query's results.
-    pub async fn execute(&self) -> Result<DatasetRecordBatchStream> {
+    pub async fn into_stream(&self) -> Result<DatasetRecordBatchStream> {
```

Comment:
nit: `into_` methods will take (owned) `self` not `&self`. (see: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)

What's wrong with `.execute()`? IMO that seemed like a fine name.

Reply:
`execute` means the materialization happens within the function itself (as it is done already after the call finishes).

`something_become_a_stream()` means that it returns a lazy evaluated stream that users should take care buffered / try_collect and etc. ?

Reply:
what about `to_stream()` or `stream()`? Any other suggestions?

Reply:
But `to` / `into` mean converting from one type to some similar type. I think someone holding a `Query` object is looking for the word `execute`, because they want to do that action (execute the query). Whether or not the full results are materialized within the functions seems not so relevant IMO.

In DataFusion, they call the method `execute_stream()` [^1]

[^1]: https://docs.rs/datafusion/latest/datafusion/dataframe/struct.DataFrame.html#method.execute_stream

Reply:
ok, changed to execute_stream()